x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmSettingsTests.java

```diff
@@ -0,0 +1,374 @@
+            );
+        }
+        for (final String ignoredValue : new String[] { null }) {
+            final Settings settings = Settings.builder().put(settingKey, ignoredValue).build();
+            final String actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(setting.getDefault(settings)));
+        }
+        for (final String acceptedValue : new String[] { "sharedsecret", "none" }) {
+            final Settings settings = Settings.builder().put(settingKey, acceptedValue).build();
+            final String actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(acceptedValue));
+        }
+    }
+
+    public void testTimeSettingsWithDefault() {
+        for (final Setting.AffixSetting<TimeValue> setting : List.of(
+            JwtRealmSettings.ALLOWED_CLOCK_SKEW,
+            JwtRealmSettings.CACHE_TTL,
+            JwtRealmSettings.HTTP_CONNECTION_READ_TIMEOUT,
+            JwtRealmSettings.HTTP_SOCKET_TIMEOUT
+        )) {
+            final String settingKey = RealmSettings.getFullSettingKey(REALM_NAME, setting);
+            for (final String rejectedValue : new String[] { "", "-2", "10", "1w", "1M", "1y" }) {
+                final Exception exception = expectThrows(IllegalArgumentException.class, () -> {
+                    final Settings settings = Settings.builder().put(settingKey, rejectedValue).build();
+                    final TimeValue actualValue = this.buildConfig(settings).getSetting(setting);
+                    fail("No exception. Expected one for " + settingKey + "=" + rejectedValue + ". Got " + actualValue + ".");
+                });
+                assertThat(
+                    exception.getMessage(),
+                    equalTo(
+                        "failed to parse setting ["
+                            + settingKey
+                            + "] with value ["
+                            + rejectedValue
+                            + "] as a time value: unit is missing or unrecognized"
+                    )
+                );
+            }
+            for (final String ignoredValue : new String[] { null }) {
+                final Settings settings = Settings.builder().put(settingKey, ignoredValue).build();
+                final TimeValue actualValue = this.buildConfig(settings).getSetting(setting);
+                assertThat(actualValue, equalTo(setting.getDefault(settings)));
+            }
+            for (final String acceptedValue : new String[] { "-1", "0", "0s", "1s", "1m", "1h", "1d" }) {
+                final Settings settings = Settings.builder().put(settingKey, acceptedValue).build();
+                final TimeValue actualValue = this.buildConfig(settings).getSetting(setting);
+                assertThat(actualValue, equalTo(TimeValue.parseTimeValue(acceptedValue, settingKey)));
+            }
+        }
+    }
+
+    public void testIntegerSettingsWithDefault() {
```

Comment:
Do we need to parse generic settings parsing code in this class ?

Reply:
I included it for completeness over all JWT realm settings. We don't need to test Integer parser functionality, only test the completeness of our settings to parse certain values as an integer.

Reply:
I'm not suggesting that we add more tests, I'm questioning if we need this test at all. I think we don't